nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -497,6 +498,20 @@ public Description matchAssignment(AssignmentTree tree, VisitorState state) {
       GenericsChecks.checkTypeParameterNullnessForAssignability(tree, this, state);
     }
 
+    if (config.isJSpecifyMode() && tree.getVariable() instanceof ArrayAccessTree) {
+      ArrayAccessTree arrayAccess = (ArrayAccessTree) tree.getVariable();
+      ExpressionTree arrayExpr = arrayAccess.getExpression();
+      ExpressionTree expression = tree.getExpression();
+      Symbol arraySymbol = ASTHelpers.getSymbol(arrayExpr);
+      if (arraySymbol != null) {
+        boolean isElementNullable = isArrayElementNullable(arraySymbol, config);
+        if (!isElementNullable && mayBeNullExpr(state, expression)) {
+          String message = "assigning @Nullable expression to @NonNull field.";
+          return buildDescription(tree).setMessage(message).build();
```

Comment:
We need a more appropriate message here, like `"Writing @Nullable expression into array with @NonNull contents"`.  Also you should use the `ErrorBuilder` APIs and add a new `ErrorMessage` message type.

Reply:
Fixed!